Source/qol/floatinginfobox.cpp

```diff
@@ -0,0 +1,485 @@
+	DUR
+};
+
+enum ItemRequirementType {
+	STR,
+	DEX,
+	MAG
+};
+
+std::map<ItemStatType, UiFlags> GetItemBonusColors(const Item &item)
+{
+	std::map<ItemStatType, UiFlags> itemBonusColors;
+
+	itemBonusColors[DAM] = UiFlags::ColorWhite;
+	itemBonusColors[AC] = UiFlags::ColorWhite;
+	itemBonusColors[DUR] = UiFlags::ColorWhite;
+
+	ItemData itemData = AllItemsList[item.IDidx];
+
+	// Here we set the color for the numbers that follow Armor:, Damage:, and Durability:
+	if (item._iMagical == ITEM_QUALITY_UNIQUE) {
+		const UniqueItem &uitem = UniqueItems[item._iUid];
+		assert(uitem.UINumPL <= sizeof(uitem.powers) / sizeof(*uitem.powers));
+		for (const auto &power : uitem.powers) {
+			switch (power.type) {
+			case IPL_ACP:
+				itemBonusColors[AC] = UiFlags::ColorBlue;
+				break;
+			case IPL_ACP_CURSE:
+				itemBonusColors[AC] = UiFlags::ColorRed;
+				break;
+			case IPL_SETAC:
+				if (item._iAC > itemData.iMaxAC)
+					itemBonusColors[AC] = UiFlags::ColorBlue;
+				else if (item._iAC < itemData.iMinAC)
+					itemBonusColors[AC] = UiFlags::ColorRed;
+				else
+					itemBonusColors[AC] = UiFlags::ColorWhite;
+				break;
+			case IPL_DAMMOD:
+			case IPL_DAMP:
+			case IPL_TOHIT_DAMP:
+				itemBonusColors[DAM] = UiFlags::ColorBlue;
+				break;
+			case IPL_DAMP_CURSE:
+			case IPL_TOHIT_DAMP_CURSE:
+				itemBonusColors[DAM] = UiFlags::ColorRed;
+				break;
+			case IPL_SETDAM:
+				if ((item._iMinDam > itemData.iMinDam) || (item._iMaxDam > itemData.iMaxDam))
+					itemBonusColors[DAM] = UiFlags::ColorBlue;
+				else if ((item._iMinDam < itemData.iMinDam) || (item._iMaxDam < itemData.iMaxDam))
+					itemBonusColors[DAM] = UiFlags::ColorRed;
+				break;
+			case IPL_DUR:
+				itemBonusColors[DUR] = UiFlags::ColorBlue;
+				break;
+			case IPL_DUR_CURSE:
+				itemBonusColors[DUR] = UiFlags::ColorRed;
+				break;
+			case IPL_SETDUR:
+				if (item._iMaxDur > itemData.iDurability)
+					itemBonusColors[DUR] = UiFlags::ColorBlue;
+				else if (item._iMaxDur < itemData.iDurability)
+					itemBonusColors[DUR] = UiFlags::ColorRed;
+				break;
+			default:
+				break;
+			}
+		}
+	} else if (item._iMagical == ITEM_QUALITY_MAGIC) {
+		switch (item._iPrePower) {
+		case IPL_ACP:
+			itemBonusColors[AC] = UiFlags::ColorBlue;
+			break;
+		case IPL_DAMP:
+		case IPL_TOHIT_DAMP:
+			itemBonusColors[DAM] = UiFlags::ColorBlue;
+			break;
+		case IPL_ACP_CURSE:
+			itemBonusColors[AC] = UiFlags::ColorRed;
+			break;
+		case IPL_DAMP_CURSE:
+		case IPL_TOHIT_DAMP_CURSE:
+			itemBonusColors[DAM] = UiFlags::ColorRed;
+			break;
+		default:
+			break;
+		}
+
+		switch (item._iSufPower) {
+		case IPL_DAMMOD:
+			itemBonusColors[DAM] = UiFlags::ColorBlue;
+			break;
+		case IPL_DUR:
+			itemBonusColors[DUR] = UiFlags::ColorBlue;
+			break;
+		case IPL_DUR_CURSE:
+			itemBonusColors[DUR] = UiFlags::ColorRed;
+			break;
+		default:
+			break;
+		}
+	}
+
+	return itemBonusColors;
+}
+
+std::map<ItemRequirementType, UiFlags> GetItemRequirementColors(const Item &item)

```

Comment:
Completely stupid way to handle this. Should just refactor this function so I can call it and specify which exact thing I want item requirement color for (STR, DEX, or MAG) and have it return what I need instead of having it return all 3 and having to write them to a new map variable.